base/loading.jl

```diff
@@ -526,6 +526,15 @@ evaluated by `julia -e <expr>`.
 """
 macro __DIR__() source_dir() end
 
+"""
+    current_location() -> Int
+
+Return the line number where the current macro expansion was invoked.
+"""
+function current_location()
+    convert(Int, ccall(:jl_current_lineno, Cint, ()))
```

Comment:
Other than making this thread local, shouldn't this just be `Int(unsafe_load(cglobal(:jl_lineno, Cint)))`?